crates/wick/wick-runtime/src/triggers/http/service_factory.rs

```diff
@@ -0,0 +1,173 @@
+  if let Err(e) = stream {
+    return Ok(
+      Builder::new()
+        .status(StatusCode::INTERNAL_SERVER_ERROR)
+        .body(Body::from(e.to_string()))
+        .unwrap(),
+    );
+  }
+  let mut stream = stream.unwrap();
+  let mut builder = Response::builder();
+  let mut body = bytes::BytesMut::new();
+  while let Some(packet) = stream.next().await {
+    match packet {
+      Ok(p) => {
+        if p.port_name() == "response" {
+          if p.is_done() {
+            continue;
+          }
+          let response: wick_interface_http::HttpResponse = p
+            .deserialize()
+            .map_err(|e| HttpError::Deserialize("response".to_owned(), e.to_string()))?;
+          builder = convert_response(builder, response)?;
+        } else if p.port_name() == "body" {
+          if p.is_done() {
+            continue;
+          }
+          let response: bytes::Bytes = p
+            .deserialize()
+            .map_err(|e| HttpError::Deserialize("body".to_owned(), e.to_string()))?;
+          body.extend_from_slice(&response);
+        }
+      }
+      Err(e) => return Err(HttpError::InvalidResponse(e.to_string())),
+    }
+  }
+  Ok(builder.body(body.freeze().into()).unwrap())
+}
+
+impl Service<Request<Body>> for ResponseService {
+  type Response = Response<Body>;
+  type Error = hyper::Error;
+  type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + Sync>>;
+
+  fn poll_ready(&mut self, _: &mut Context) -> Poll<Result<(), Self::Error>> {
+    Poll::Ready(Ok(()))
+  }
+
+  fn call(&mut self, req: Request<Body>) -> Self::Future {
+    trace!("http:trigger:request");
+    let network = self.network.clone();
+    let router = self
+      .routers
+      .iter()
+      .find(|r| req.uri().path().starts_with(&r.path))
+      .cloned();
+    trace!(?router, "http:trigger:request");
+
+    Box::pin(async move {
+      let handler = match router {
+        Some(h) => {
+          let invocation = Invocation::new(
+            Entity::client("http_client"),
+            Entity::operation(&h.component, &h.operation),
+            None,
+          );
+          match convert_request(req) {
+            Ok((req, mut body)) => {
+              let packets = packets!(("request", req));
+              let (tx, rx) = PacketStream::new_channels();
+              for packet in packets {
+                let _ = tx.send(packet);
+              }
+              while let Some(bytes) = body.next().await {
+                match bytes {
+                  Ok(b) => {
+                    let _ = tx.send(Packet::encode("body", b));
+                  }
+                  Err(e) => {
+                    let _ = tx.send(Packet::err("body", e.to_string()));
+                  }
+                }
+              }
+              let _ = tx.send(Packet::done("body"));
+              let stream = network.invoke(invocation, rx);
```

Comment:
This should probably be invoked earlier. As it is, this code waits for the entire body to be read before invoking the operation.